Adds logic to the petskill "first" variable within OnPetSkillFinished
This variable is used to determine if a mob has been interacted with for skillchaining, as even AoE skills can only SC a single mob

This patch abuses that variable and sets it to false if the pet has a player owner and that owner does not have avatar's favor active
---
diff --git a/src/map/entities/pet_entity.cpp b/src/map/entities/pet_entity.cpp
index a65ef61e01..bad7ff7d80 100644
--- a/src/map/entities/pet_entity.cpp
+++ b/src/map/entities/pet_entity.cpp
@@ -29,6 +29,7 @@
 #include "ai/states/petskill_state.h"
 #include "data/enums/mob_mod.h"
 #include "packets/pet_sync.h"
+#include "status_effect_container.h"
 #include "utils/battleutils.h"
 #include "utils/messageutils.h"
 #include "utils/mobutils.h"
@@ -499,7 +500,14 @@ void CPetEntity::OnPetSkillFinished(CPetSkillState& state, action_t& action)
     MsgBasic msg            = MsgBasic::None;
     MsgBasic defaultMessage = PSkill->getMsg();
 
-    bool first{ true };
+    bool canSkillchain = true;
+    if (auto* PChar = dynamic_cast<CCharEntity*>(this->PMaster); PChar && this->getPetType() == PET_TYPE::AVATAR)
+    {
+        // If pet is an avatar and pet's owner is a PC, lock skillchaining behind avatar's favor
+        canSkillchain = PChar->StatusEffectContainer->HasStatusEffect(xi::StatusEffect::AvatarsFavor);
+    }
+    // this bool is only ever used for skillchain. If in the future it gets used for something else this logic will need to be adjusted.
+    bool first{ canSkillchain };
     for (auto&& PTargetFound : PAI->TargetFind->m_targets)
     {
         action_target_t& actionTarget = action.addTarget(PTargetFound->id);
